src/map/skill.cpp

```diff
@@ -15759,12 +15759,33 @@ int32 skill_castend_map (map_session_data *sd, uint16 skill_id, const char *mapn
 
 static bool skill_dance_switch(struct skill_unit* unit, bool revert);
 
-/// transforms 'target' skill unit into dissonance (if conditions are met)
+/**
+ * Turn a dissonance-overlapped skill unit back to its original form
+ * @param unit: Unit to transform back
+ */
+void skill_dance_overlap_revert(skill_unit* unit) {
+	// Need to delete previous unit on the client as it can't handle unit_id changes
+	clif_skill_delunit(*unit);
+	// Remove overlap signal
+	unit->val2 &= ~(1 << UF_ENSEMBLE);
+	// If the unit is removed because overlap dissonance killed the caster, we need to reset it here
+	skill_dance_switch(unit, true);
+	skill_getareachar_skillunit_visibilty(unit, AREA);
+}
+
+/**
+ * Transforms overlapping skill units into dissonance (if conditions are met)
+ * @param bl: Target skill unit found that is on the same cell as source skill unit
+ * @param ap: func* with va_list values:
+ *  src - source skill unit
+ *  flag - Whether to remove dissonance, add dissonance or just return if the overlap conditions are met (OVERLAP_COUNT)
```

Comment:
```suggestion
 *  flag - see enum e_dance_overlap
```

Reply:
I wanted to expand on the enum explanation here a little bit so developers can see what the flag means in the context of this function. You think it is sufficient to just link to the enum?